course_info/tests/jasmine/controllers/PeopleControllerSpec.js

```diff
@@ -128,19 +160,540 @@ describe('Unit testing PeopleController', function() {
+            scope.dtInstance = {reloadData: function(){}};
+            spyOn(scope.dtInstance, 'reloadData');
+
+            // call it
+            scope.addUserToCourse(searchTerm, user);
+
+            // trigger the ajax calls
+            $httpBackend.expectPOST(coursePeopleURL, user)
+                        .respond(201, JSON.stringify(partialFailureResponse)); 
+            $httpBackend.expectGET(enrollmentDetailsURL)
+                        .respond(200, JSON.stringify(enrollmentDetails));
+            $httpBackend.flush(2);
+
+            // check to see if it's reacting correctly
+            expect(scope.successes).toEqual([expectedSuccess]);
+            expect(scope.partialFailures).toEqual([expectedPartialFailure]);
+            expect(scope.dtInstance.reloadData.calls.count()).toEqual(1);
+        });
+
+        it('should warn on failure to add the user', function() {
+            spyOn(scope, 'handleAjaxError');
+
+            scope.addUserToCourse(searchTerm, user);
+
+            $httpBackend.expectPOST(coursePeopleURL, user).respond(500, '');
+            $httpBackend.flush(1);
+
+            expect(scope.handleAjaxError.calls.count()).toEqual(1);
+            expect(scope.warnings).toEqual([{type: 'addFailed',
+                                             searchTerm: searchTerm}]);
+        });
+
+        it('should handle a failure to get user enrollment after an apparent add success',
+           function() {
+               var expectedPartialFailure = {
+                   searchTerm: searchTerm,
+                   text: 'Add to course seemed to succeed, but we received ' +
+                         'an error trying to retrieve the user\'s course details.',
+               };
+               spyOn(scope, 'handleAjaxError');
+
+               scope.addUserToCourse(searchTerm, user);
+
+               $httpBackend.expectPOST(coursePeopleURL, user)
+                           .respond(201, JSON.stringify(user)); 
+               $httpBackend.expectGET(enrollmentDetailsURL).respond(404, '');
+               $httpBackend.flush(2);
+
+               expect(scope.partialFailures).toEqual([expectedPartialFailure]);
+               expect(scope.handleAjaxError.calls.count()).toEqual(1);
+           }
+        );
+    });
+    describe('handleLookupResults', function() {
+        // NOTE: relies on filterResults() working properly.  mocking
+        //       its results wasn't worth it.
+        beforeEach(function() {
+            var ci = {
+                course_instance_id: $routeParams.courseInstanceId,
+                title: 'handleLookupResults test',
+            };
+            courseInstances.instances[ci.course_instance_id] = ci;
+            controller = $controller('PeopleController', {$scope: scope});
+            scope.searchInProgress = true;
+        });
+
+        it('should warn and disable progress if the user is already enrolled',
+           function() {
+               var peopleResult = {};
```

Comment:
maybe a small comment like you've done above that this var/content doesn't matter would be helpful.